### PR TITLE
docs(sample): relocate native image sample from old repo

### PR DESCRIPTION
**Pull request by mpeddada1**
Relocating Spanner sample from [GoogleCloudPlatform/native-image-support-java](https://github.com/GoogleCloudPlatform/native-image-support-java/tree/master/native-image-samples/native-image-samples-client-library/spanner-sample). 
This PR also adds an integration test that can be run as a native image.

Calling `mvn package -Pnative -DskipTests` builds the native image for the application and calling `mvn test -Pnative` runs the test as a native image. 

For more information: https://graalvm.github.io/native-build-tools/latest/maven-plugin.html#configuration.

cc @ansh0l @meltsufin

### CODE REVIEW

**Review by ansh0l — APPROVED** [review]
LGTM. 

Minor nit: I think since the parent folder is named `samples`, we should call this one as `native-image` instead of `native-image-sample`

**Comment by mpeddada1** [comment]
Thank you @ansh0l! I've renamed this to `native-image`

**Comment by ansh0l** [comment]
Closing since native image changes have been released with https://github.com/googleapis/java-spanner/pull/1878

**Review by ansh0l — CHANGES_REQUESTED** [review]
@mpeddada1 : Are you looking to merge this PR? it has conflicts in its readme, please check.

**Comment by mpeddada1** [comment]
@ansh0l, thank you, I've resolved the merge conflict.

**Comment by mpeddada1** [comment]
All checks are now green.

File: samples/native-image/pom.xml [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    <maven.compiler.source>1.8</maven.compiler.source>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+  </properties>
+
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>com.google.cloud</groupId>
+        <artifactId>libraries-bom</artifactId>
+        <version>25.2.0</version>
```
Comment (ansh0l) [inline samples/native-image/pom.xml:31]: QQ> AFAIK, currently, we get automated PRs for cloud bom version bumps. Would now there be 2 such PRs (one for regular samples/pom.xml, another for samples/native-image/pom.xml), or would both end up coming into 1 PR?

**Review by ansh0l — COMMENTED** [review]
At a quick glance, looks alright. Please take a full review with @rajatbhatta

File: samples/native-image/README.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
+# Spanner Sample Application with Native Image
+
+This is a sample application which uses the Cloud Spanner client libraries and demonstrates compatibility with Native Image native image compilation.
```
Comment (rajatbhatta) [inline samples/native-image/README.md:3]:
```suggestion
This is a sample application which uses the Cloud Spanner client libraries and demonstrates compatibility with Native Image compilation.
```

Comment (mpeddada1) [reply samples/native-image/README.md]: Good catch! Done.

**Review by rajatbhatta — APPROVED** [review]
LGTM.

**Pull request closed without merging**